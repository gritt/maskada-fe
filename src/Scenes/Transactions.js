import React from 'react'
import {Transaction} from "../Components/Transaction";

function Transactions(props) {
    return (
        <section className="content">
            <ul>
                {props.transactions.map(function (transaction) {
                    return (<Transaction {...transaction} key={transaction.id}/>);
                })}
            </ul>
        </section>
    );
}

export {Transactions}