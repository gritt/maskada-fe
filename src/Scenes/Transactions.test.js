import React from 'react';
import renderer from 'react-test-renderer';

import {Transactions} from "./Transactions";

const testTransactions = {
    "January 20": [{
        "amount": 300,
        "category": "Health",
        "date": "2020-01-25T16:35:13Z",
        "id": 2,
        "name": "",
        "type": 2
    }],
    "October 19": [{
        "amount": 4050,
        "category": "Income",
        "date": "2019-10-23T16:35:13Z",
        "id": 1,
        "name": "",
        "type": 3
    }]
};

const testMonth = "January 20";

describe("Transactions", () => {
    it("renders scene with empty list", () => {
        // given
        const given = renderer.create(
            <Transactions transactions={[]} activeMonth={undefined}></Transactions>
        );

        // when
        const got = given.toJSON();

        // then
        expect(got).toMatchSnapshot();
    });
    it('renders scene with a list of transactions of the active month', () => {
        // given
        const given = renderer.create(
            <Transactions
                transactions={testTransactions}
                activeMonth={testMonth}>
            </Transactions>
        );

        // when
        const got = given.toJSON();

        // then
        expect(got).toMatchSnapshot();
    });
});