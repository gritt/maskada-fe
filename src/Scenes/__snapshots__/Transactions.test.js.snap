// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions renders scene with a list of transactions of the active month 1`] = `
<section
  className="content"
>
  <ul>
    <li
      className="transaction"
    >
      <span
        className="transaction__description"
      >
        <span
          className="transaction__name"
        >
          Health
        </span>
        <span
          className="transaction__date"
        >
          2020-01-25T16:35:13Z
        </span>
      </span>
      <span>
        <span
          className="transaction__amount transaction__type-credit"
        >
          $ 
          300
        </span>
        <span
          className="transaction__connection"
        />
      </span>
    </li>
  </ul>
</section>
`;

exports[`Transactions renders scene with empty list 1`] = `
<section
  className="content"
>
  <i
    className="content__no-transactions"
  >
    "if I don't buy anything, discount is bigger..."
  </i>
</section>
`;
